### ISSUE TITLE
EFITEquilibrium returns negative psi_normalised close to magnetic axis

### ISSUE THREAD

**Issue by Mateasek** [cherab/core#147]
It recently happened to me that I got a negative psi_normalised close to the magnetic axis. This is imo caused by the Interpolate2DCubic function used. This could be solved by using ClampOutput1D with bottom bound equal to 0.

**Comment by Mateasek** [comment]
Another issue there might be is the way the psi_normalised interpolator is generated:
``` python

self.psi_normalised = Interpolate2DCubic(r, z, (psi - psi_axis) / (psi_lcfs - psi_axis))
```
if there is not the coordinate for magnetic axis in the rz mesh, then the interpolator will not get 0 psin, but some finite positive number which will act as an incorrect interpolation limit, if then by chance you hit closer to the magnetic axis, while mapping profiles, you get an interpolation error.

This could be solved by allowing the user to switch the extrapolation to True and choose the type. This could solve both issues. Any other ideas?

**Comment by Mateasek** [comment]
I absolutely confused myself... :-D

**Comment by Mateasek** [comment]
I am reopening this issue because I've encountered the problem again. I loaded an equilibrium generated by fiesta code from eqdsk file. Then I sampled the region close to the magnetic axis and several obtained values of psi normalised were indeed negative. This I think should nod be allowed as it is against definition of the psi normalised. I think the 2d interpolator output should be clamped:

```python
self.psi_normalised = ClampOutput2D(Interpolate2DCubic(r, z, (psi - psi_axis) / (psi_lcfs - psi_axis)), min=0)
```

I believe this is connected to the interpolator oscillations I've encountered previously.
